File: 백준/Silver/12847. 꿀 아르바이트/꿀 아르바이트.js
//const input = require("fs").readFileSync("input.txt", "utf-8").split("\n");
const input = require("fs").readFileSync("/dev/stdin", "utf-8").split("\n");

const [n, m] = input[0].trim().split(" ").map(Number);
const t = input[1].trim().split(" ").map(Number);

let start = 0;
let end = 0;
let max_pay = -1;

let prefix = [0];
for (let i = 0; i < n; i++) {
	prefix.push(prefix[i] + t[i]);
}

while (true) {
	if (end === n) {
		break;
	}
	if (end - start <= m) {
		const total = prefix[end] - prefix[start];
		max_pay = Math.max(max_pay, total);
		end++;
	} else {
		start++;
	}
}

console.log(max_pay);

File: 백준/Silver/12847. 꿀 아르바이트/꿀 아르바이트.py
n,m = map(int,input().split())
t = list(map(int,input().split()))

start = 0
end = 0
max_pay = -1

prefix = [0]
for i in range(n):
    prefix.append(prefix[i] + t[i])

while True:
    if end == n:
        break
    if end - start <= m:
        total = prefix[end] - prefix[start]
        max_pay = max(max_pay, total)
        end += 1 
    else:
        start += 1
print(max_pay)